contracts/PriceOracle/PriceOracle.sol

```diff
@@ -110,34 +102,36 @@ contract PriceOracle is Ownable2Step {
         // Initialize the aggregator to read the price from
         AggregatorV3Interface priceFeed = AggregatorV3Interface(config.priceFeed);
         // Retrieve decimals from feed for formatting
-        uint8 decimals = priceFeed.decimals();
-        // Fail safe check since this is an unrealistic scenario. The number of digits that the int256 answer can have naturally limits decimals
-        // but this protects against an erroneous uint8 values
-        if (decimals > MAX_DECIMALS) revert DecimalsTooLarge(decimals);
+        uint8 feedDecimals = priceFeed.decimals();
+        // Cap the sum of feed and underlying asset decimals at 72 to avoid overflows when formatting prices
+        // Unrealistic scenario but added as a fail safe
+        // Comptroller expects 0 price on error.
+        if (feedDecimals + config.underlyingAssetDecimals > MAX_DECIMALS) return 0;
         // Retrieve price from feed
         (
             /* uint80 roundID */,
             int256 answer,
-            /*uint startedAt*/,
-            /*uint timeStamp*/,
+            /*uint256 startedAt*/,
+            uint256 updatedAt,
             /*uint80 answeredInRound*/
         ) = priceFeed.latestRoundData();
-        if (answer < 0) revert NegativePrice(answer);
+        // Price considered stale since it has not been updated within the staleness window. Comptroller expects 0 price on error.
+        if (block.timestamp > updatedAt + STALE_PRICE_DELAY_SECONDS) return 0;
```

Comment:
Have we checked that this is acceptable with Compound?

Reply:
I had not but I will. I think this would need to be set in line with the feeds' heartbeat configs which I was thinking they wouldn't have an insight into.

Reply:
With the feedback from Compound and OpenZeppelin, I ended up removing the staleness check from the contract. Returning a 0 price could do more harm to the protocol than returning a stale price so we're removing this validation.